perceval/runtime/remote_processor.py

```diff
@@ -148,13 +148,18 @@ def set_parameter(self, key: str, value: Any):
             warn(f"'{key}' parameter is deprecated. Use `remote_processor.noise = NoiseModel(...)` instead.",
                  DeprecationWarning)
 
-    def set_circuit(self, circuit: ACircuit):
+    def check_circuit(self, circuit: ACircuit):
         if 'max_mode_count' in self.constraints and circuit.m > self.constraints['max_mode_count']:
             raise RuntimeError(f"Circuit too big ({circuit.m} modes > {self.constraints['max_mode_count']})")
         if 'min_mode_count' in self.constraints and circuit.m < self.constraints['min_mode_count']:
             raise RuntimeError(f"Circuit too small ({circuit.m} < {self.constraints['min_mode_count']})")
-        if self._input_state is not None and self._input_state.m != circuit.m:
-            raise RuntimeError(f"Circuit and input state size do not match ({circuit.m} != {self._input_state.m})")
+        if self._input_state is not None:
```

Comment:
If you talk about this whole method, well no because `self.constraints` is not defined for local `Processor`.
If you talk about the size check between input state and circuit, it's already done.

Reply:
I talk about the size check between input state and circuit, you not factor the code and put it in the parent class then ?